src/api/controllers/voteController.ts

```diff
@@ -0,0 +1,62 @@
+import { BaseAPI } from "../abstract";
+import {
+  CreateVoteResponse,
+  CreateVoteRequest,
+  PartialVoteRequest,
+} from "./types/voteController.type";
+
+export class VoteController extends BaseAPI {
+  private baseUrl = "https://api.thecatapi.com/v1";
+
+  private headers() {
+    if (!process.env.API_KEY) {
+      throw new Error("API_KEY is not defined");
+    }
+    return {
+      "x-api-key": process.env.API_KEY,
+      "Content-Type": "application/json",
+    };
+  }
+
+  async addVote(
+    data: CreateVoteRequest | PartialVoteRequest
```

Comment:
```suggestion
    data: PartialVoteRequest
```
since the `PartialVoteRequest` contains everything from the `CreateVoteRequest`

Reply:
Fixed